extensions/amp-subscriptions/0.1/viewer-subscription-platform.js

```diff
@@ -102,25 +104,33 @@ export class ViewerSubscriptionPlatform {
       }
     }
     if (encryptedDocumentKey) {
-      messageData['encryptedDocumentKey'] = encryptedDocumentKey;
+      authRequest['encryptedDocumentKey'] = encryptedDocumentKey;
     }
-    const entitlementPromise = this.viewer_
-      .sendMessageAwaitResponse('auth', messageData)
+
+    return /** @type {!Promise<Entitlement>} */ (this.viewer_
+      .sendMessageAwaitResponse('auth', authRequest)
       .then(entitlementData => {
-        const authData = (entitlementData || {})['authorization'];
-        const decryptedDocumentKey = (entitlementData || {})[
-          'decryptedDocumentKey'
-        ];
+        entitlementData = entitlementData || {};
+
+        const error = entitlementData['error'];
+        const authData = entitlementData['authorization'];
+        const decryptedDocumentKey = entitlementData['decryptedDocumentKey'];
+
+        if (error) {
```

Comment:
We don't want this error to be sent to the viewer?

Reply:
This error would have come from the viewer, so sending it back seemed redundant